### PR TITLE
puppeteer: Upgrade typings for version 0.13

### PR DESCRIPTION
**Pull request by ghost**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [v0.13.0](https://github.com/GoogleChrome/puppeteer/releases/tag/v0.13.0)
- [x] Increase the version number in the header if appropriate.

Additional changes:
- JSDoc added

### CODE REVIEW

**Comment by marvinhagemeister** [comment]
Awesome, that looks quite thorough! Will have a look at it tomorrow (it's 2am here). Thanks already for your efforts, much appreciated 👍

**Comment by knuppe** [comment]
Everything should be fine now! :ok_hand:

File: types/puppeteer/index.d.ts [hunk 1]
```diff
@@ -66,6 +139,12 @@ export type PageEvents =
   | "requestfinished"
   | "response";
 
+export type BrowserEvents =
```
Comment (marvinhagemeister) [inline types/puppeteer/index.d.ts]: I can't find any place where these are used. Perhaps this was a leftover of something else?

File: types/puppeteer/index.d.ts [hunk 2]
```diff
@@ -346,90 +729,364 @@ export interface FrameBase {
...
+  /**
+   * Wait for the page navigation occur.
+   * @param options The navigation parameters.
+   */
   waitForNavigation(options?: NavigationOptions): Promise<Response>;
 }
 
+/** A Browser is created when Puppeteer connects to a Chromium instance, either through puppeteer.launch or puppeteer.connect. */
 export interface Browser {
+  on<K extends keyof BrowserEventObj>(
```
Comment (marvinhagemeister) [inline types/puppeteer/index.d.ts]: The [docs](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md#class-browser) don't mention an `on()`-method. Perhaps their document is outdated, or this doesn't exist.

**Review by marvinhagemeister — COMMENTED** [review]
Only a few minor nitpicks, otherwise really great!

@typescript-bot Well this was a bit premature! /cc @andy-ms How should we proceed?

**Comment by sandersn** [comment]
Sorry, @typescript-bot interprets :+1: on comments as signoff. And I hardly look at "Express"-tagged reviews before merging. @knuppe are you able to make a followup PR addressing @marvinhagemeister's comments?

**Comment by marvinhagemeister** [comment]
@sandersn That's good to know, should've been more careful with my emojis :) No worries, these are only minor nitpicks and can easily be fixed in a follow-up PR.

File: types/puppeteer/index.d.ts (hunk 2 above)
Comment (knuppe) [reply types/puppeteer/index.d.ts]: The docs mention the events (disconnected, targetchanged, targetcreated and the targetdestroyed). The `Browser` class [extends](https://github.com/GoogleChrome/puppeteer/blob/master/lib/Browser.js#L21) the [EventEmitter](https://nodejs.org/api/events.html) that has the `on` method.

I think we should be fine.

File: types/puppeteer/index.d.ts (hunk 1 above)
Comment (knuppe) [reply types/puppeteer/index.d.ts]: The `BrowserEvents` have the same purpose as the `PageEvents`, they are just references to available events in the `Browser` and `Page` classes. They can be used to make dynamic event handlers.

Technically, neither type is required, I just kept the code standard from the previous version.

File: types/puppeteer/index.d.ts (hunk 2 above)
Comment (marvinhagemeister) [reply types/puppeteer/index.d.ts]: You're right, thanks for clearing this up! Just had a look at the source and it indeed does extend event emitter. :radar:

File: types/puppeteer/index.d.ts (hunk 1 above)
Comment (marvinhagemeister) [reply types/puppeteer/index.d.ts]: Makes sense. :+1:

**Pull request merged**